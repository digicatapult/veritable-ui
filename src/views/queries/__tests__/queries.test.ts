import { expect } from 'chai'
import { describe, it } from 'mocha'
import QueriesTemplates from '../queries.js'

describe('QueriesTemplates', () => {
  describe('chooseQueryPage', () => {
    it('should render queries to select', async () => {
      const templates = new QueriesTemplates()
      const rendered = await templates.chooseQueryPage()
      expect(rendered).to.matchSnapshot()
    })
  })
})
