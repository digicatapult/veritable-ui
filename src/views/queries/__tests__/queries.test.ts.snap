// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionTemplates listPage should render with no connections 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx-ext-json-enc/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon-dark.ico\\" media=\\"(prefers-color-scheme: dark)\\"/><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon.ico\\" media=\\"(prefers-color-scheme: light), (prefers-color-scheme: no-preference)\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/query.css\\"/><title>Veritable - Queries</title></head><body hx-ext=\\"json-enc\\"><nav id=\\"side-bar\\"><img src=\\"/public/images/logo-square.svg\\"/><div class=\\"profile-icon\\" style=\\"background-color:#38b6ff;\\">A</div><a title=\\"categories\\" href=\\"/\\" style=\\"--background-image:url('/public/images/category.svg');\\"></a><a title=\\"connections\\" href=\\"/connection\\" style=\\"--background-image:url('/public/images/connection.svg');\\"></a><a title=\\"queries\\" href=\\"/queries\\" data-active style=\\"--background-image:url('/public/images/query.svg');\\"></a><a title=\\"credentials\\" href=\\"/credentials\\" style=\\"--background-image:url('/public/images/credential.svg');\\"></a><a title=\\"settings\\" href=\\"#\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/setting.svg');\\"></a></nav><main><div id=\\"content-header\\"><h1>Queries</h1><div id=\\"content-header-nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><a title=\\"Query Management\\" href=\\"/queries\\">Query Management</a><a title=\\"New\\" href=\\"/queries/new\\">New</a></div></div><div id=\\"content-main\\"><div class=\\"main-list-page\\" hx-get=\\"/queries/new\\"></div><div><h1 class=\\"query-page-header\\">Queries</h1><hr class=\\"divider\\"/></div><div class=\\"query-container\\"><a class=\\"query-item\\" href=\\"/queries/new/scope-3-carbon-consumption\\"><h1 class=\\"query-header\\">Total Carbon Embodiment</h1><p class=\\"query-text\\">Creates a query for calculating the total carbon embodiment for a given product or component.</p></a><a class=\\"query-item disabled\\" href=\\"#\\"><h1 class=\\"query-header\\">Product Provenance</h1><p class=\\"query-text\\">Creates a query to assert that a product and its components are not coming from blacklisted areas.</p></a><a class=\\"query-item  disabled\\" href=\\"#\\"><h1 class=\\"query-header\\">Aerospace Grade</h1><p class=\\"query-text\\">Creates a query to certify all components of a product are manufactured according to aerospace grade standards such as AS9100.</p></a><a class=\\"query-item  disabled\\" href=\\"#\\"><h1 class=\\"query-header\\">ISO 9001 Compliance</h1><p class=\\"query-text\\">Creates a query to verify that all the companies involved in manufacturing a given product are ISO9001 certified.</p></a></div></div></main></body></html>"`;
